juju/juju
# feat: expose multi-model agent service

On the controller service factory, expose an Agent() service that can retrieve agent version information for arbitrary models. The service was already defined in the domain package, it just wasn't available to downstream users via the service factory.

The current (model-scoped) Agent() service has been renamed to ModelAgent().

We also add a `ControllerAgentVersion` method to the service for future convenience.

This will be needed e.g. in the Controller facade, which needs to access the agent version for migrating models during migration prechecks.

## Checklist




## QA steps

This is an internal change. Run unit tests. Ensure modified facades (application, agenttools) continue to work as expected.

## Links

**Jira card:** JUJU-6589
